diagnostic/engine.go

```diff
@@ -12,18 +12,113 @@
 // See the License for the specific language governing permissions and
 // limitations under the License.
 
-package inference
+// Package diagnostic hosts the diagnostic engine, which is responsible for collecting the
+// conflicts from annotation-based checks (no-infer mode) and/or inference (full-infer mode) and
+// generating user-friendly diagnostics from those conflicts.
+package diagnostic
```

Comment:
This file is moved from `inference/conflict.go`. Please see `inference/conflict.go` for the updates to the file content.